server/src/main/java/org/apache/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -103,11 +106,10 @@
   private final SQLMetadataConnector connector;
 
   // Volatile since this reference is reassigned in "poll" and then read from in other threads.
-  // Starts null so we can differentiate "never polled" (null) from "polled, but empty" (empty map).
```

Comment:
Why was this comment removed? It looks still valid.

Reply:
ah removed it when I removed the dataSources map, but it's still relevant if `DataSourcesSnapshot` has empty dataSources map and empty overshadowedSegments. Added the comment.